pkg/kubelet/server/server.go

```diff
@@ -194,10 +194,17 @@ func ListenAndServeKubeletServer(
 
 	if tlsOptions != nil {
 		s.TLSConfig = tlsOptions.Config
-		// Passing empty strings as the cert and key files means no
-		// cert/keys are specified and GetCertificate in the TLSConfig
-		// should be called instead.
-		if err := s.ListenAndServeTLS(tlsOptions.CertFile, tlsOptions.KeyFile); err != nil {
+		certFile := tlsOptions.CertFile
```

Comment:
I believe a better place to clear the Cert/KeyFile would be at the spot where the `GetCertificate()` function is being set: https://github.com/kubernetes/kubernetes/blob/e1773f1afd16d7e2d9e0cd9b2c853734abb76ea5/pkg/kubelet/kubelet.go#L913-L921

Here it may not be clear where `tlsOptions` come from, and if `tlsOptions.CertFile` and `tlsOptions.KeyFile` would also be returned in a `GetCertificate()` call.
On the other hand, if we clear these fields in the above mentioned code, all consumers should continue to use all of `Cert/KeyFile` and `config.GetCertificate()` for their serving configuration.

Reply:
Yeah, I didn't do this initially because I thought it might be confusing to have `tlsOptions.CertFile` and `tlsOptions.KeyFile` set in `InitializeTLS`, and then unset again later in `NewMainKubelet`. But I agree it might be less error prone to unset them immediately.

I've added the change, (+ a rebase onto `master`)